matbench/task.py

```diff
@@ -315,6 +336,37 @@ def record(self, fold_number, predictions, params=None):
             ids_to_predictions = {split_ids[i]: p for i, p in enumerate(predictions)}
             self.results[fold_key][self._DATA_KEY] = ids_to_predictions
 
+            if std is not None or ci is not None:
+                if self.metadata["task_type"] == "classification":
+                    raise ValueError(
+                        "`std` and `ci` are not valid kwargs for classification tasks. See https://github.com/materialsproject/matbench/pull/99/files#issuecomment-1022662192."
+                    )
+
+                if ci is None:
+                    low_p = 0.05
+                    high_p = 0.95
+                    # convert from two-tail to one-tail probabilities for compatibility with `ppf` #https://stackoverflow.com/a/29562808/13697228
+                    low_p = low_p / 2.0
+                    high_p = (1 + high_p) / 2.0
+                    # convert std to ci, modified from source:
+                    # https://github.com/uncertainty-toolbox/uncertainty-toolbox/blob/b2f342f6606d1d667bf9583919a663adf8643efe/uncertainty_toolbox/metrics_scoring_rule.py#L187
+                    pred_l = stats.norm.ppf(low_p, loc=predictions, scale=std)
+                    pred_u = stats.norm.ppf(high_p, loc=predictions, scale=std)
+                    ci = np.vstack((pred_l.ravel(), pred_u.ravel())).T.tolist()
+                    ci = [tuple(c) for c in ci]
+
+                if len(ci) != len(split_ids):
+                    raise ValueError(
+                        f"Confidence interval outputs (derived from standard deviations if `std` was supplied) must be the same length as the "
+                        f"inputs! {len(ci)} != {len(split_ids)}"
+                    )
+
+                ids_to_uncertainties = {
+                    split_ids[i]: {"ci_lower": p[0], "ci_upper": p[1]}
+                    for i, p in enumerate(ci)
+                }
+                self.results[fold_key][self._UNCERTAINTY_KEY] = ids_to_uncertainties
```

Comment:
`ids_to_uncertainties` and storing results

Reply:
could we store the std as well? just so looking at an individual sample we get all the info without having to reverse engineer

Reply:
@ardunn (not sure how notifications work for these kinds of conversations)

Reply:
Hmm this is an interesting question. 

I think for sanity's sake we should just store it as `None` if it's not specified and not symmetric. But we should have some logic in `validate` that makes sure each entry is consistent. For example, if std is specified and ci is specified for one sample, it must be specified for all samples. If ci is specified but std is not, that must be consistent for all samples. Does that seem reasonable to you?

Reply:
@ardunn I agree with storing it as `None`. Also agreed about the entries being consistent (assuming sample means a single MP compound). Seems like an easy enough check to implement.

If it's not specified, but it is symmetric within a small numerical tolerance, I think I'll convert this to `std` so that it's easy to tell which uncertainties are Gaussian or not without the user needing to specify whether it is or not. Sound OK?

Reply:
Yes sounds good!

Reply:
@ardunn I think these points should be addressed now. The `validate` logic may not be the most elegant. Let me know if it captures what you described. Also, the linting should be OK now too. Thanks for pointing to the linting script.

Reply:
@ardunn do you want me to modify the test JSON file to match what is expected for the uncertainty syntax?